### PR TITLE
Add collection for users to database #139

### LINKED ISSUE #139: Add collection for users to database
As a developer I want to have a collection for all registerd user to fetch relevant data

acceptanse critera:

- [ ] A user object will contain properties:
- displayName
- email

### LINKED ISSUE #141: When adding a new user to collection 'User' set reference to currentUser.uid
As a developer I want the reference to a document in collection 'user' to be the same as the users ID in auth.

Solution: https://stackoverflow.com/questions/48541270/how-to-add-document-with-custom-id-to-firestore

### PR DESCRIPTION
**Pull request by liinneea-a**
closes #139 
closes #141 

This PR will:
- Add input for display name when signing up.
- save all new registered user to db collection 'users'.
- fix so when logged in user updates name display name also updates in db

### CODE REVIEW

**Review by AdaHep — APPROVED** [review]
Nicely done, much cleaner code than before and works great as far as i can tell.

File: src/Pages/ProfilePage.tsx [hunk 1]
```diff
@@ -69,7 +77,7 @@ function ProfilePage() {
               alt="profile picture"
             />
           )}
-          {currentUser.displayName ? (
+          {/* {currentUser.displayName ? ( */}
```
Comment (mirrenil) [inline src/Pages/ProfilePage.tsx:80]: This should be removed :)

File: src/Pages/SignUp.tsx [hunk 2]
```diff
@@ -9,35 +9,34 @@ import { useFormik } from "formik";
...
+  const { signup, googleSignIn } = useAuth();
+  const [email, setEmail] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [confirmationPassword, setConfirmationPassword] = useState<string>("");
+  const [displayName, setDisplayedName] = useState<string>("");
 
-  const handleSignup = async () => {
-    if (passwordRef.current?.value !== passwordConfirmationRef.current?.value) {
-      return console.log("Passwords do not match");
+  const handleSubmit = async () => {
```
Comment (mirrenil) [inline src/Pages/SignUp.tsx:24]: Can you please rename this to handleSignUp as it was in main? :)

**Review by hazem-89 — DISMISSED** [review]

**Review by mirrenil — CHANGES_REQUESTED** [review]
Kan du byta tillbaka till handleSignUp? :)

File: src/Pages/SignUp.tsx (hunk 2 above)
Comment (liinneea-a) [reply src/Pages/SignUp.tsx:24]: yez

**Review by hazem-89 — APPROVED** [review]
looks good

**Review by mirrenil — APPROVED** [review]

**Review by hazem-89 — APPROVED** [review]

**Pull request merged**